src/v/redpanda/admin/migrations.cc

```diff
@@ -104,102 +104,66 @@ void write_migration_as_json(
 json::validator make_migration_validator() {
     const std::string schema = R"(
 {
+    "$schema": "https://json-schema.org/draft/2020-12/schema#",
     "type": "object",
-    "properties": {
-        "migration_type": {
-            "description": "Migration type",
-            "type": "string",
-            "enum": [
-                "inbound",
-                "outbound"
-            ]
-        }
-    },
-    "required": [
-        "migration_type"
-    ],
-    "allOf": [
+    "anyOf": [
```

Comment:
I think you can reduce the scope of this `anyOf`, and I think it should be `oneOf`:
```json
    "topics": {
      "description": "Topics to migrate",
      "type": "array",
      "items": {
        "oneOf": [
          { "$ref": "#/definitions/inbound_topic" },
          { "$ref": "#/definitions/namespaced_topic" }
        ]
      }
    },
```

Reply:
> I think you can reduce the scope of this `anyOf`
Ignore me, I somehow had the wrong version of the schema.

Reply:
I expect in-/outbound migration definitions to further diverge in future, and it's going to be top-level attributes of migrations themselves, not topics. Internal migration definition objects already have extra properties like this.